plugins/outputs/influxdb/client/http.go

```diff
@@ -233,15 +233,24 @@ func (c *httpClient) makeWriteRequest(
 		return nil, err
 	}
 	req.Header.Set("Content-Length", fmt.Sprint(contentLength))
```

Comment:
This is going to complicate things... if we are using gzip we need to report the length of the body here after compression, but since we are now streaming the data it is trickier.  I think we need to use `Transfer-Encoding: chunked` and this header is unset for gzip.

I haven't ever done this with go, maybe we can use https://golang.org/pkg/net/http/httputil/#NewChunkedWriter?  Or maybe it will be automatically handled if we don't set this header?